test/test_producer.py

```diff
@@ -116,6 +122,8 @@ def test_kafka_producer_proper_record_metadata(kafka_broker, compression):
 
     assert record.serialized_key_size == 10
     assert record.serialized_value_size == 12
+    if headers:
+        assert record.serialized_header_size == 22
```

Comment:
Can we check exact header data here. I want to be sure we have str as keys and bytes as values.

Reply:
The record here refers to the FutureRecordMetadata / RecordMetadata that doesn't carry the actual stored values.

I've verified this manually for both Python 3.6 and 2.7 with a producer & consumer running through actual Kafka service instance. The types are str, bytes for 3.6 and unicode, str/bytes for 2.7.